ci/dhis2-verify-app.yml

```diff
@@ -0,0 +1,172 @@
+name: 'dhis2: verify (app)'
+
+on:
+    push:
+        branches:
+            # match branches in:
+            # https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches
+            - master
+            - next
+            - next-major
+            - alpha
+            - beta
+            - '[0-9]+.x'
+            - '[0-9]+.x.x'
+            - '[0-9]+.[0-9]+.x'
+
+    pull_request:
```

Comment:
Does this work on to push feature branches to d2-ci when on a pull request branch?  I assume it does...

Reply:
> > install job runs first, blocking parallel jobs (test, lint) until the cache has been created.
> 
> Were you able to test this? I'm not sure we gain much by saving and restoring the Yarn cache between jobs, as the install job itself still needs to run for each job and now it actually needs to run twice before any job can finish. Maybe we could cache `node_modules` in the `install` job and drop the `yarn install` repeated for each other job instead?

Yeah, I tested it and it works in some capacity, but in cli-style an install is so fast that it just saves 4 seconds (from 5 seconds to 1 second) so it is negligible. The "win" would really be in application workflow runs.

I like the idea of caching the node_modules, we can try that instead.

> Alternatively, just having the cache action and `yarn install` for each of the `lint`/`test`/`build`/`publish` jobs gains a fair amount since the cache hit happens between workflow runs, not just between jobs, and avoids needing an `install` job at all

Yeah, I thought about this too. The majority of our pull requests are Dependabot PRs, and those are going to be cache misses since they always (well ...) change the `yarn.lock` file.

So in those cases the parallel jobs would always run install `n` times, where `n` is the number of jobs in parallel. And each of the Dependabot PRs that rebases gets a new `yarn.lock` as well and triggers `n` runs, and so forth for all the open Dependabot PRs.

We can easily have 10:1 of Dependabot PRs vs. Human PRs, so I figure we should optimize for the 10x case if we want to save build minutes and :evergreen_tree:.

If you disagree with this rationale, we can go for omitting the "install" job completely.